scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py

```diff
@@ -0,0 +1,190 @@
+
+        cmd = f"test -e /dev/vg_metadata_{node_name}/lv_main_swap"
+        self.execute_command(cmd, TIMEOUT)
+
+        cmd = f"swapon /dev/vg_metadata_{node_name}/lv_main_swap"
+        self.execute_command(cmd, TIMEOUT)
+
+        cmd = (
+            f"echo \"/dev/vg_metadata_{node_name}/lv_main_swap    swap    "
+            f"swap    defaults        0 0\" >> /etc/fstab"
+        )
+        self.execute_command(cmd, TIMEOUT)
+
+
+    def config_lvm(self):
+        node_name = self.conf_store.get(self.index, f'cluster>server[{self.server_id}]>hostname', default_val=None)
+        node_name = node_name.split('.')[0]
+        metadata_device = self.conf_store.get(self.index,
+        f'cluster>server[{self.server_id}]>storage>metadata_devices', default_val=None)
+        is_physical = True if self.conf_store.get(self.index, f'cluster>server[{self.server_id}]>node_type', default_val=None) == "HW" else False
+
+        self.create_lvm(node_name, metadata_device[0], is_physical)
+
+    def config_motr(self):
+        is_physical = True if self.conf_store.get(self.index, f'cluster>server[{self.server_id}]>node_type', default_val=None) == "HW" else False
+        if is_physical:
+            cmd = f"/opt/seagate/cortx/motr/libexec/motr_cfg.sh"
+            self.execute_command(cmd, TIMEOUT)
+
+    def validate_motr_rpm(self):
+        cmd = f"uname -r"
+        op = self.execute_command(cmd, TIMEOUT)
+        kernel_ver = op.replace('\n', '')
+
+        self.validate_file(f"/lib/modules/{kernel_ver}/kernel/fs/motr/m0tr.ko")
+
+        self.validate_file("/etc/sysconfig/motr")
+
+    def validate_file(self, file):
+        if not os.path.exists(file):
+            sys.stderr.write(f"[ERR] {file} does not exist.\n")
+            sys.exit(1)
+        else:
+            sys.stdout.write(f"[MSG] {file} exists.\n")
+
+    def test_lnet(self):
+        missing_pkgs = []
+        LNET_CONF_FILE="/etc/modprobe.d/lnet.conf"
+        search_lnet_pkgs = ["kmod-lustre-client", "lustre-client"]
+
+        # Check missing luster packages
+        cmd = "rpm -qa | grep lustre"
+        temp =  self.execute_command(cmd, TIMEOUT)
+        lustre_pkgs = list(filter(None, temp.split("\n")))
+        for pkg in search_lnet_pkgs:
+            found = False
+            for lustre_pkg in lustre_pkgs:
+                if pkg in lustre_pkg:
+                    found = True
+                    break;
+            if found == False:
+                missing_pkgs.append(pkg)
+        if missing_pkgs:
+            sys.stderr.write("Missing pkgs ={}\n".format(missing_pkgs))
+
+        # Check for lnet config file
+        if os.path.exists(LNET_CONF_FILE):
+            with open(LNET_CONF_FILE) as fp:
+                line = fp.readline()
+                while line:
+                    tokens = line.split(' ')
+                    # Get lnet iface
+                    cmd = 'echo \'{}\' | cut -d "(" -f2 | cut -d ")" -f1'.format(tokens[2])
```

Comment:
There is no need to run shell command for this. Python is better in parsing. Use "re" package. 

```
def get_lnet_xface():
    lnet_xface = None
    try:
        with open('/etc/modprobe.d/lnet1.conf', 'r') as f:
            # Obtain interface name
            for line in f.readlines():
                if len(line.strip()) <= 0: continue
                tokens = re.split(r'\W+', line)
                if len(tokens) < 5: continue
                lnet_xface = tokens[4]
    except:
        pass

    if lnet_xface == None:
        raise MotrError(errno.EINVAL, "Cant obtain iface details from %s",
            '/etc/modprobe.d/lnet.conf')
    if lnet_xface not in os.listdir('/sys/class/net/'):
        raise MotrError(errno.EINVAL, "Invalid iface %s in lnet.conf",
            lnet_xface)
```